### PR TITLE
Rust: Various fixes to the CFG construction

### PR DESCRIPTION
**Pull request by paldepind**
This PR fixes a bunch of inconsistencies in the CFG construction. This should reduce the number of CFG inconsistencies we see on real projects quite a bit.

Most of these are simple fixes and come with test cases and separate commits, so I won't describe them in detail. Please ask or comment if there's anything that's not clear.

### CODE REVIEW

**Comment by paldepind** [comment]
According to the DCA results this PR seems to fix the vast majority of CFG inconsistencies. There's a few still left in some projects. I'll take a look at those as well and update the PR.

File: rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll [hunk 1]
```diff
@@ -612,6 +612,8 @@ module PatternTrees {
       result = rank[i + 1](Pat pat, int j | pat = this.getPat(j) | pat order by j)
     }
 
+    predicate isEmpty() { not any(Pat p) = this.getPat(0) }
```
Comment (hvitved) [inline rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll]: `not exists(this.getPat(_))`.

**Review by hvitved — CHANGES_REQUESTED** [review]
LGTM, only one minor thing. Thanks a lot for structuring the PR into easy reviewable commits.

File: rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll (hunk 1 above)
Comment (paldepind) [reply rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll]: True. I realized that that was a bug, fixed in a later commit and after https://github.com/github/codeql/pull/17800/commits/6568eb80a2bbe77beb044af52afa13bb96b55208 it got removed as the behavior is now inherited from `StandardTree`.

**Comment by paldepind** [comment]
I've updated the PR to fix the CI and add one more inconsistency fix (`match` expressions without arms) that I think should resolve the remaining inconsistencies that the DCA report showed.

**Review by hvitved — APPROVED** [review]

**Pull request merged**